### PR TITLE
feat: hardware wallet confirmation screen

### PR DESCRIPTION
**Pull request by bfullam**
## **Description**

Bridge currently redirects to swap when trying to use a hardware wallet. This PR allows hardware wallet accounts to access swaps and adds an info screen to inform the user about the transactions they will need to approve on their device.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/29113?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Connect with hardware wallet account
2. Go to bridge
3. Enter token, amount, and network
4. Confirm quote
5. See info screen

### **After**
Screenshot 2024-12-11 at 20 52 29

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/bridge/awaiting-signatures/bridge-step-icon.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function BridgeStepIcon({ stepNumber = 1 }) {
```
Comment (micaelae) [inline ui/pages/bridge/awaiting-signatures/bridge-step-icon.tsx:4]: Minor: [AvatarBase](https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-avatarbase--docs) can be used to show the numbers as well

File: ui/pages/bridge/index.tsx [hunk 2]
```diff
@@ -93,41 +96,51 @@ const CrossChainSwap = () => {
...
+                <PrepareBridgePage />
+              </Content>
+              <Footer>
+                <BridgeCTAButton />
+              </Footer>
+            </>
+          </FeatureToggledRoute>
+
+          <Route path={CROSS_CHAIN_SWAP_ROUTE + AWAITING_SIGNATURES_ROUTE}>
+            <Content className="bridge__content">
```
Comment (micaelae) [inline ui/pages/bridge/index.tsx:136]: You can remove the className here since there's no scss style for it

Comment (bfullam) [reply ui/pages/bridge/index.tsx]: Done

File: ui/pages/bridge/awaiting-signatures/bridge-step-icon.tsx (hunk 1 above)
Comment (bfullam) [reply ui/pages/bridge/awaiting-signatures/bridge-step-icon.tsx]: Thanks for the tip, done!

File: ui/pages/bridge/awaiting-signatures/awaiting-signatures.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  const fromChain = useSelector(getFromChain, isEqual);
+  const toChain = useSelector(getToChain, isEqual);
+  const hardwareWalletUsed = useSelector(isHardwareWallet);
+  const hardwareWalletType = useSelector(getHardwareWalletType);
+  const needsTwoConfirmations = Boolean(activeQuote?.approval);
+  const trackEvent = useContext(MetaMetricsContext);
+
+  useEffect(() => {
+    trackEvent({
+      event: 'Awaiting Signature(s) on a HW wallet',
```
Comment (jclancy93) [inline ui/pages/bridge/awaiting-signatures/awaiting-signatures.tsx:51]: I don't think this event is in our metrics spec

File: ui/pages/bridge/awaiting-signatures/awaiting-signatures.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      category: MetaMetricsEventCategory.Swaps,
+      sensitiveProperties: {
```
Comment (jclancy93) [inline ui/pages/bridge/awaiting-signatures/awaiting-signatures.tsx:53]: Same things apply here as your mobile PR. Token addresses, need two confs and hardware wallet metadata is ok to include in properties, just token amounts will need to be marked as sensitive

File: ui/pages/bridge/awaiting-signatures/awaiting-signatures.tsx (hunk 3 above)
Comment (bfullam) [reply ui/pages/bridge/awaiting-signatures/awaiting-signatures.tsx:51]: This event is copied from the matching "awaiting-signatures" file in swaps: https://github.com/MetaMask/metamask-extension/blob/0265469dddd94533d4f1522cc17620d7120f7c28/ui/pages/swaps/awaiting-signatures/awaiting-signatures.js#L61

**Review by jclancy93 — DISMISSED** [review]
Code and E2E Tests LGTM. Followed up async about a question around event tracking

File: ui/pages/bridge/awaiting-signatures/awaiting-signatures.tsx (hunk 4 above)
Comment (bfullam) [reply ui/pages/bridge/awaiting-signatures/awaiting-signatures.tsx]: Fixed!

**Review by jclancy93 — DISMISSED** [review]

**Review by ghgoodreau — DISMISSED** [review]

**Review by ghgoodreau — APPROVED** [review]

**Review by jclancy93 — APPROVED** [review]

**Review by micaelae — APPROVED** [review]

**Pull request merged**